internal/server/dataset.go

```diff
@@ -438,79 +438,142 @@ func (ds *Dataset) StoreEntities(entities []*Entity) (Error error) {
 							return err
 						}
 
-						oldRefs[predid] = relatedid
+						if refs, ok := oldRefs[predid]; ok {
+							refs = append(refs, relatedid)
+							oldRefs[predid] = refs
+						} else {
+							refs := []uint64 { relatedid}
+							oldRefs[predid] = refs
+						}
```

Comment:
I think you can append to unitialized slices. like this should work:
```
 var a []string
a = append(a, "hi") 
```
So maybe we don't need the if/else